src/Sulu/Bundle/WebsiteBundle/Command/DumpSitemapCommand.php

```diff
@@ -120,16 +135,33 @@ private function dumpSitemap(PortalInformation $portalInformation)
             $portalInformation->setUrl(str_replace('{host}', $this->defaultHost, $portalInformation->getUrl()));
         }
 
-        if (!$this->sitemapProviderPool->needsIndex()) {
+        $filePath = $this->sitemapRenderer->getIndexDumpPath(
+            $this->scheme,
+            $portalInformation->getWebspaceKey(),
+            $portalInformation->getLocale(),
+            $portalInformation->getHost()
+        );
+
+        $sitemap = $this->sitemapRenderer->renderIndex($portalInformation->getHost(), $this->scheme);
+        if (!$sitemap) {
+            $aliases = array_keys($this->sitemapProviderPool->getProviders());
+
             $this->dumpFile(
-                '/' . $portalInformation->getUrl() . '/sitemap.xml',
-                $this->renderSitemap($this->sitemapProviderPool->getFirstAlias(), 1, $portalInformation)
+                $filePath,
+                $this->sitemapRenderer->renderSitemap(
+                    reset($aliases),
+                    1,
+                    $portalInformation->getLocale(),
+                    $portalInformation->getPortal(),
+                    $portalInformation->getHost(),
+                    $this->scheme
+                )
             );
 
             return;
         }
 
-        $this->dumpFile('/' . $portalInformation->getUrl() . '/sitemap.xml', $this->renderIndex($portalInformation));
+        $this->dumpFile($filePath, $sitemap);
```

Comment:
Should we maybe move that to the consecutive `foreach` loop? Then there wouldn't be a `sitemap.xml` with wrong contents in case one of the providers produces an error.